molgenis-ontology-core/src/main/java/org/molgenis/ontology/utils/NgramGenerator.java

```diff
@@ -0,0 +1,274 @@
+			}
+		}
+		return stringBuilder.toString().trim();
+	}
+
+	private List<String> createLinkedNGrams(String inputQuery)
+	{
+		Set<String> wordsInString = new HashSet<String>(Arrays.asList(inputQuery.trim().split(" ")));
+		List<String> tokens = new ArrayList<String>();
+		// Padding the string
+		for (String singleWord : wordsInString)
+		{
+			if (!StringUtils.isEmpty(singleWord))
+			{
+				// The s$ will be the produced from two words.
+				StringBuilder singleString = new StringBuilder(singleWord.length() + 2);
+				singleString.append('^').append(singleWord.toLowerCase()).append('$');
+				int length = singleString.length();
+				for (int i = 0; i < length - 1; i++)
+				{
+					if (i + nGrams < length) tokens.add(singleString.substring(i, i + nGrams));
+					else tokens.add(singleString.substring(length - 2));
+				}
+			}
+		}
+		return tokens;
+	}
+
+	private String generateRandomString(int length)
+	{
+		StringBuilder stringBuilder = new StringBuilder();
+		String currentToken = drawTokenOnPossibility(startNGramTokenFreq, 1);
+		if (StringUtils.isEmpty(currentToken)) return StringUtils.EMPTY;
+		stringBuilder.append(currentToken);
+		stringBuilder.delete(1, stringBuilder.length());
+
+		for (int i = 0; i < length - 1; i++)
+		{
+			if (currentToken.endsWith("$")) break;
+
+			double factor = 0;
+			Map<String, Double> tempMap = new HashMap<String, Double>();
+			for (String possibleToken : nGramRelations.get(currentToken))
+			{
+				factor += nGramTokenFreq.get(possibleToken);
+				tempMap.put(possibleToken, nGramTokenFreq.get(possibleToken));
+			}
+			currentToken = drawTokenOnPossibility(tempMap, factor);
+			if (StringUtils.isEmpty(currentToken)) return StringUtils.EMPTY;
+			stringBuilder.delete(stringBuilder.length() - 1, stringBuilder.length()).append(currentToken);
+		}
+
+		if (stringBuilder.toString().endsWith("$")) stringBuilder.delete(stringBuilder.length() - 1,
+				stringBuilder.length());
+
+		while (stringBuilder.length() != length)
+		{
+			stringBuilder.delete(0, stringBuilder.length()).append(generateRandomString(length));
+		}
+
+		return stringBuilder.toString();
+	}
+
+	private String drawTokenOnPossibility(Map<String, Double> nGramTokenFreq, double factor)
+	{
+		double p = Math.random() * factor;
+		double cumulativeProbability = 0.0;
+		for (Entry<String, Double> entry : nGramTokenFreq.entrySet())
+		{
+			cumulativeProbability += entry.getValue();
+			if (p <= cumulativeProbability)
+			{
+				return entry.getKey();
+			}
+		}
+		return null;
+	}
+
+	public static boolean isSignificantVal(List<Double> listOfValues, double expectedValue)
+	{
+		double[] sample = new double[listOfValues.size()];
+		for (int i = 0; i < listOfValues.size(); i++)
+		{
+			sample[i] = listOfValues.get(i);
+		}
+
+		double tTest = TestUtils.tTest(expectedValue, sample);
+		return tTest < T_TEST_ALPHA;
+	}
+
+	public static void main(String args[]) throws OWLOntologyCreationException
```

Comment:
What is the goal of this "main" function, added voor debug?
Please remove commented code